File: dynabuffers-java/src/test/kotlin/dynabuffers/usecase/Schema05Test.kt
package dynabuffers.usecase

import dynabuffers.AbstractDynabuffersTest
import dynabuffers.Dynabuffers
import org.junit.jupiter.api.Test

class Schema05Test : AbstractDynabuffersTest() {

    @Test
    fun testParse() {
        val engine = Dynabuffers.parse(Schema05Test::class.java.getResourceAsStream("/schema05.dbs"))
        val output = mapOf("strings" to arrayOf("a", "b", "c"))

        assertMap(engine, mapOf("strings" to listOf("a", "b", "c")), output)
    }

}

File: dynabuffers-python/tests/usecase/Schema05Test.py
import os
import unittest

from antlr4 import FileStream

from dynabuffers.Dynabuffers import Dynabuffers


class Schema05Test(unittest.TestCase):
    root_dir = os.path.dirname(os.path.realpath(__file__))

    def test_parse(self):
        engine = Dynabuffers.parse(FileStream(self.root_dir + "/schema05.dbs"))
        map = engine.deserialize(engine.serialize({"strings" : ["a", "b", "c"]}))

        self.assertEqual(map, {"strings" : ["a", "b", "c"]})


if __name__ == "__main__":
    unittest.main()
